homeassistant/components/sensor/homematicip.py

```diff
@@ -0,0 +1,418 @@
+            return 'Access Point Status'
+        return '{} Access Point Status'.format(self._home.label)
+
+    @property
+    def icon(self):
+        """Return the icon of the access point device."""
+        return 'mdi:access-point-network'
+
+    @property
+    def state(self):
+        """Return the state of the access point."""
+        return self._home.dutyCycle
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the access point."""
+        return {
+            ATTR_LABEL: self._home.label,
+            ATTR_ID: self._home.id,
+            ATTR_CONNECTED: self._home.connected,
+            ATTR_DUTYCYCLE: self._home.dutyCycle,
+            ATTR_CURRENT_APVERSION: self._home.currentAPVersion,
+            ATTR_AVAILABLE_APVERSION: self._home.availableAPVersion,
+            ATTR_TIMEZONE_ID: self._home.timeZoneId,
+            ATTR_PIN_ASSIGNED: self._home.pinAssigned,
+            ATTR_UPDATE_STATE: self._home.updateState,
+            ATTR_APEXCHANGE_CLIENTID: self._home.apExchangeClientId,
+            ATTR_APEXCHANGE_STATE: self._home.apExchangeState,
+            }
+
+
+class HomematicipDeviceStatus(HomematicipGenericDevice):
+    """Representation of an HomematicIP sensor device."""
+
+    def __init__(self, hass, home, device, signal=None):
+        """Initialize the device."""
+        super().__init__(hass, home, device)
+        _LOGGER.debug('Setting up sensor device status: %s',
+                      device.label)
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name('Status')
+
+    @property
+    def icon(self):
+        """Return the icon of the generic device."""
+        if self._device.unreach is None or self._device.unreach:
+            return 'mdi:wifi-off'
+        elif self._device.lowBat:
+            return 'mdi:battery-outline'
+        elif (hasattr(self._device, 'sabotage') and
+              self._device.sabotage == HMIP_SABOTAGE):
+            return 'mdi:alert'
+        elif self._device.updateState.lower() != HMIP_UPTODATE:
+            return 'mdi:refresh'
+        return 'mdi:check'
+
+    @property
+    def state(self):
+        """Return the state of the generic device."""
+        if self._device.unreach is None or self._device.unreach:
+            return ATTR_UNREACHABLE
+        elif self._device.lowBat:
+            return ATTR_LOW_BATTERY
+        elif (hasattr(self._device, 'sabotage') and
+              self._device.sabotage == HMIP_SABOTAGE):
+            return ATTR_SABOTAGE
+        elif self._device.updateState.lower() != HMIP_UPTODATE:
+            return self._device.updateState.lower()
+        return 'OK'
+
+    def _sensor_state_attributes(self):
+        """Return the state attributes of the generic device."""
+        attr = self._generic_state_attributes()
+
+        if hasattr(self._device, 'on'):
+            attr.update({ATTR_ON: self._device.on})
```

Comment:
This is wrong. If it is an on/off sensor it should be a binary sensor. Make this a binary sensor platform.

Reply:
I'm stopping now with reviewing, lots of comments already to address.

Reply:
I think I fixed most of the topics you have highlighted and got idea behind.